packages/headless/src/controllers/static-filter/headless-static-filter.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  staticFilterIdSchema,
+  staticFilterValuesSchema,
+} from '../../features/static-filter-set/static-filter-set-schema';
+import {
+  deselectAllStaticFilterValues,
+  registerStaticFilter,
+  toggleSelectStaticFilterValue,
+} from '../../features/static-filter-set/static-filter-set-actions';
+import {StaticFilterValue} from '../../features/static-filter-set/static-filter-set-state';
+import {StaticFilterSection} from '../../state/state-sections';
+import {loadReducerError} from '../../utils/errors';
+import {validateOptions} from '../../utils/validate-payload';
+
+const optionsSchema = new Schema<Required<StaticFilterOptions>>({
+  id: staticFilterIdSchema,
+  values: staticFilterValuesSchema,
+});
+
+export interface StaticFilterProps {
+  /**
+   * The options for the `StaticFilter` controller.
+   */
+  options: StaticFilterOptions;
+}
+
+export interface StaticFilterOptions {
+  /**
+   * A unique identifier for the static filter.
+   */
+  id: string;
+
+  /**
+   * The values the static filter is responsible for managing.
+   */
+  values: StaticFilterValue[];
+}
+
+/**
+ * The `StaticFilter` controller manages a collection of filter values.
+ * */
+export interface StaticFilter extends Controller {
+  /**
+   * Toggles the specified static filter value.
+   *
+   * @param value - The static filter value to toggle.
+   */
+  toggleSelect(value: StaticFilterValue): void;
+
+  /**
+   * Toggles the specified static filter value, deselecting others.
+   *
+   * @param value - The static filter value to toggle.
+   */
+  toggleSingleSelect(value: StaticFilterValue): void;
+
+  /**
+   * Deselects all static filter values.
+   * */
+  deselectAll(): void;
+
+  /**
+   * A state of the `StaticFilter` controller.
+   */
+  state: StaticFilterState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `StaticFilter` controller.
+ */
+export interface StaticFilterState {
+  /**
+   * The static filter id.
+   */
+  id: string;
+
+  /**
+   * The static filter values.
+   */
+  values: StaticFilterValue[];
+
+  /**
+   * `true` if there is at least one non-idle value and `false` otherwise.
+   */
+  hasActiveValues: boolean;
+}
+
+/**
+ * Creates a `Static Filter` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `Sort` controller properties.
+ * @returns A `Sort` controller instance.
+ */
+export function buildStaticFilter(
+  engine: SearchEngine,
+  props: StaticFilterProps
+): StaticFilter {
+  if (!loadReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  validateOptions(engine, optionsSchema, props.options, 'buildStaticFilter');
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+  const {id} = props.options;
+
+  dispatch(registerStaticFilter(props.options));
+
+  return {
+    ...controller,
+
+    toggleSelect(value: StaticFilterValue) {
```

Comment:
I think it would be nice to have a boolean `selected` parameter or something of the sort. It could just exit the function is the current state matches the parameter.